# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1209140264 14400
#      Fri Apr 25 12:17:44 2008 -0400
# Node ID 6848970153ba5e2e1b59284bd258d91abd2e0e96
# Parent  2dee19385d32f869a2bd0a2b6cd3de5a108850c9
do not use output value of fill_n (not supported by MSVC).

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,13 @@
+2008-04-25  Michael Goffioul <michael.goffioul@gmail.com>
+
+	* oct-lookup.h (seq_lookup): Do not use output value of fill_n (MSVC
+	does not support it).
+
 2008-04-24  Michael Goffioul  <michael.goffioul@gmail.com>
 
 	* lo-sysdep.cc (octave_popen2): Don't set PIPE_NOWAIT for parentWrite.
 
 2008-04-21  John W. Eaton  <jwe@octave.org>
 
 	* idx-vector.cc (IDX_VEC_REP::idx_vector_rep (const boolNDArray&)):
 	Fix for-loop condition.
diff --git a/liboctave/oct-lookup.h b/liboctave/oct-lookup.h
--- a/liboctave/oct-lookup.h
+++ b/liboctave/oct-lookup.h
@@ -127,17 +127,18 @@ seq_lookup (const T *table, octave_idx_t
               vnew = std::find_if (vcur, vend,
                                    not1 (bind2nd (comp, *cur)));
           else
             // special case: uppermost range [max, Inf)
             vnew = std::find_if (vcur, vend,
                                  bind2nd (comp, *(cur-1)));
 
           // store index of the current interval.
-          idx = std::fill_n (idx, vnew - vcur, cur - table);
+          std::fill_n (idx, vnew - vcur, cur - table);
+	  idx += (vnew - vcur);
           vcur = vnew;
 
         }
     }
 }
 
 // overload using < operator
 template<typename T, typename bpred>
